spotify-artist-viewer/Networking/SARequestManager.m

```diff
@@ -2,22 +2,165 @@
+            artistArray = [itemArray objectForKey:@"items"];
+            
+            
+            for(itemArray in [itemArray valueForKey:@"items"]){
+                SAArtist *a = [[SAArtist alloc]init];
+                
+                NSDictionary *imgArray = [[NSDictionary alloc] init];
+                imgArray = [itemArray objectForKey:@"images"];
+                NSArray *tempimgUrl = [imgArray valueForKey:@"url"];
+                
+                //making sure artist has images
+                
+                if(tempimgUrl.count != NULL){
+                    
+                    a.imgURL = [tempimgUrl objectAtIndex:0];
+                }
+            
+                NSString *name = [itemArray valueForKey:@"name"];
+                a.artistName = name;
+                
+                NSString *uri = [itemArray valueForKey:@"uri"];
+                a.artistUri = uri;
+          
+                
+                [artists addObject:a];
+                
+                
+            }
+            
+            
+            //econest query http://developer.echonest.com/api/v4/artist/biographies?api_key=FILDTEOIK2HBORODV&id=spotify:artist:4Z8W4fKeB5YxbusRsdQVPb
+            
+            
+            success(artists);
+            
+            
+        }
+        
+        
+        
+        
+    }];
+    
+    [task resume];
+    
+    
+    
 }
 
+-(void)getBio:(NSString *) uri success:(void (^)(NSArray *bios))success failure:(void(^)(NSError *error))failure{
+     NSString *urlPrefix = @"http://developer.echonest.com/api/v4/artist/biographies?api_key=";
+
+    NSString *myapikey =@"D4YA0K8VK2VQIUVGC&id=";
+    NSString *jsonFormat = @"&format=json";
+    
+    NSString *urlwithapi = [urlPrefix stringByAppendingString:myapikey];
+    NSString *urlwithuri = [urlwithapi stringByAppendingString:uri];
+    
+    
+    NSURL *requestURL = [NSURL URLWithString:urlwithuri];
+    NSMutableArray *bios = [NSMutableArray new];
+    NSLog(@"echonesturl - %@", urlwithuri);
+    
+    NSMutableURLRequest *urlRequest = [[NSMutableURLRequest alloc] initWithURL:requestURL];
+   
+    
+    NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration];
+    NSMutableString *bioTextArray;
+    
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+    
+        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data
+                                                             options:0
+                                                               error:NULL];
+        NSLog(@"JSON - %@", json);
+        
+        if([json isKindOfClass:[NSDictionary class]]){
+        
+            NSDictionary *bioArray = [[NSDictionary alloc] init];
+            bioArray = [json objectForKey:@"response"];
+            
```

Comment:
You can combine these lines into one, NSDictionary *bioArray = [json objectForKey:@"response"];